modules/configuring-huge-pages.adoc

```diff
@@ -5,8 +5,104 @@
 [id="configuring-huge-pages_{context}"]
 = Configuring huge pages
 
-Nodes must pre-allocate huge pages used in an {product-title} cluster. Use the
-Node Tuning Operator to allocate huge pages on a specific node.
+Nodes must pre-allocate huge pages used in an {product-title} cluster.
+There are two ways of reserving huge pages: at boot time and at run time.
+Reserving at boot time increases the possibility of success because the
+memory has not yet been significantly fragmented. The Node Tuning
+Operator currently supports boot time allocation of huge pages on specific nodes.
+
+
+== At boot time
+
+.Procedure
+
+To minimize node reboots, the order of the steps below needs to be followed:
+
+. Label all nodes that need the same hugepages setting by a label.
++
+----
+$ oc label node <node_using_hugepages> node-role.kubernetes.io/worker-hp=
+----
+
+. Create a file with the following content and name it `hugepages-tuned-boottime.yaml`:
++
+[source,yaml]
+----
+apiVersion: tuned.openshift.io/v1
+kind: Tuned
+metadata:
+  name: hugepages <1>
+  namespace: openshift-cluster-node-tuning-operator
+spec:
+  profile: <2>
+  - data: |
+      [main]
+      summary=Boot time configuration for hugepages
+      include=openshift-node
+      [bootloader]
+      cmdline_openshift_node_hugepages=hugepagesz=2M hugepages=50 <3>
+    name: openshift-node-hugepages
+
+  recommend:
+  - machineConfigLabels: <4>
+      machineconfiguration.openshift.io/role: "worker-hp"
+    priority: 30
+    profile: openshift-node-hugepages
+----
+<1> Set the `name` of the Tuned resource to `hugepages`.
+<2> Set the `profile` section to allocate huge pages.
+<3> Note the order of parameters is important as some platforms support hugepages of various sizes.
+<4> Enable MachineConfigPool based matching.
+
+
+. Create the Tuned `hugepages` profile
++
+----
+$ oc create -f hugepages-tuned-boottime.yaml
+----
+
+. Create a file with the following content and name it `hugepages-mcp.yaml`:
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfigPool
+metadata:
+  name: worker-hp
+  labels:
+    worker-hp: ""
+spec:
+  machineConfigSelector:
+    matchExpressions:
+      - {key: machineconfiguration.openshift.io/role, operator: In, values: [worker,worker-hp]}
+  nodeSelector:
+    matchLabels:
+      node-role.kubernetes.io/worker-hp: ""
+----
+
+. Create the MachineConfigPool:
++
+----
+$ oc create -f hugepages-mcp.yaml
+----
+
+Given enough non-fragmented memory, all the nodes in the `worker-hp`
+MachineConfigPool should now have 50 2Mi hugepages allocated.
+
+----
+$ oc get node <node_using_hugepages> -o jsonpath="{.status.allocatable.hugepages-2Mi}"
+100Mi
+----
+
+[WARNING]
+====
+Note this functionality is currently only supported on RHCOS 8.x worker nodes. On RHEL 7.x worker nodes the
```

Comment:
s/Note this/This

s/RHCOS/{op-system-first}

s/RHEL/{op-system-base-full}

s/plugin/plug-in

Reply:
Thank you, updated.